packages/widgets/src/HeaderBar/Apps.js

```diff
@@ -84,194 +83,218 @@ Search.propTypes = {
+                flex-wrap: wrap;
+                align-content: flex-start;
+                align-items: flex-start;
+                justify-content: flex-start;
+                width: 30vw;
+                min-width: 300px;
+                max-width: 560px;
+
+                min-height: 200px;
+                max-height: 465px;
+                margin: 0 8px 8px 8px;
+
+                overflow: auto;
+                overflow-x: hidden;
+            }
+        `}</style>
+    </div>
+)
 
-            <style jsx>{`
-                div {
-                    display: flex;
-                    flex-direction: row;
-                    flex-wrap: wrap;
-                    align-content: flex-start;
-                    align-items: flex-start;
-                    justify-content: flex-start;
-                    width: 30vw;
-                    min-width: 300px;
-                    max-width: 560px;
-
-                    min-height: 200px;
-                    max-height: 465px;
-                    margin: 0 8px 8px 8px;
-
-                    overflow: auto;
-                    overflow-x: hidden;
-                }
-            `}</style>
-        </div>
-    )
-}
 List.propTypes = {
     apps: propTypes.array,
     filter: propTypes.string,
 }
 
-export default class Apps extends React.Component {
-    state = {
-        show: false,
-        filter: '',
-    }
-
-    componentDidMount() {
-        document.addEventListener('click', this.onDocClick)
-    }
+const AppMenu = ({ apps, filter, onFilterChange }) => (
+    <div data-test="headerbar-apps-menu">
+        <Card>
+            <Search value={filter} onChange={onFilterChange} />
+            <List apps={apps} filter={filter} />
+        </Card>
+
+        <style jsx>{`
+            div {
+                z-index: 10000;
+                position: absolute;
+                top: 28px;
+                right: -6px;
+                border-top: 4px solid transparent;
+            }
+        `}</style>
+    </div>
+)
+
+AppMenu.propTypes = {
+    apps: propTypes.array.isRequired,
+    onFilterChange: propTypes.func.isRequired,
+    filter: propTypes.string,
+}
 
-    componentWillUnmount() {
-        document.removeEventListener('click', this.onDocClick)
-    }
+const mutation = {
+    resource: 'i18n',
+    type: 'create',
+    data: ({ items }) => items,
+}
 
-    onDocClick = evt => {
-        if (this.elContainer && !this.elContainer.contains(evt.target)) {
-            this.setState({ show: false })
+const Apps = ({ apps }) => {
+    const [translations, setTranslations] = useState({})
+    const [show, setShow] = useState(false)
+    const [filter, setFilter] = useState('')
+
+    const handleVisibilityToggle = useCallback(() => setShow(!show), [show])
+    const handleFilterChange = useCallback(({ value }) => setFilter(value), [])
+
+    const [mutate] = useDataMutation(mutation, {
+        onComplete: translations => setTranslations(translations),
+    })
```

Comment:
You shouldn't need a separate `translations` state variable here - the mutation returns data, which can be accessed just like you would for a query.  If this is changed to a query, that becomes even more straightforward.

```js
const [mutate, { loading, error, data: translations }] = useDataMutation(mutation)
```